chain/chain/src/types.rs

```diff
@@ -474,6 +475,7 @@ pub trait RuntimeAdapter: EpochManagerAdapter + Send + Sync {
         is_new_chunk: bool,
         is_first_block_with_chunk_of_version: bool,
         state_patch: SandboxStatePatch,
+        use_flat_storage: bool,
```

Comment:
Are we going to remove it after implementing sync & catchup logic? If this is the case, can we add TODO?

If we are not, I had an impression that code can be simplified if `get_trie_for_shard` and `apply_transactions` know whether we use flat storage or not, apparently flat storage object could be just passed there.

Reply:
We are not, because flat storage will not be enabled for state-viewer. 

I'm not sure what you mean by `get_trie_for_shard` and `apply_transactions` know whether we use flat storage or not.

Reply:
Created separate issue for it https://github.com/near/nearcore/issues/7810

(It's weird that Exalate didn't pass the description to GitHub automatically as of now)